src/prompt/variable.rs

```diff
@@ -131,106 +133,70 @@ fn quoted_str(input: &str) -> IResult<&str, String, ParsingError> {
-                                                take(2usize),
-                                                |s: &str| s.chars().all(|c| c.is_ascii_hexdigit())
-                                            )
-                                        ),
-                                        move |hex| u8::from_str_radix(hex, 16).context("Invalid hex digit")
-                                    ),
-                                    |digit| {
-                                        if digit > 0x7F || digit == 0 {
-                                            return Err(
-                                                format!(
-                                                    "invalid string literal: \
-                                                     invalid escape sequence '\\x{:x}' \
-                                                     (only non-null ascii allowed)", digit)
-                                            );
-                                        }
-
-                                        Ok(digit as char)
-                                    }
-                                ),
-                                value('\n', char('n')),
-                                value('\r', char('r')),
-                                value('\t', char('t')),
-                                value('\u{08}', char('b')),
-                                value('\u{0C}', char('f')),
-                                value('\\', char('\\')),
-                                value('/', char('/')),
-                                value('"', char('"')),
-                                value('\'', char('\'')),
-                            ))
-                        ),
-                        StringFragment::EscapedChar
-                    ),
-                    value(
-                        StringFragment::EscapedWS,
-                        preceded(char('\\'), nom::character::streaming::multispace1)
-                    ),
+        map_res(
+            recognize(
+                tuple((
+                    char(quote),
+                    move |str: &'a str| {
+                        let mut pos = 0;
+
+                        if str.is_empty() {
+                            return Err(Error(ParsingError::Incomplete));
+                        }
+
+                        if str.len() == 1 && str.chars().nth(0).unwrap() != quote {
+                            return Err(Error(ParsingError::Mistake {
+                                kind: None,
+                                description: "Missing end quote on string".to_string()
+                            }))
+                        }
+
+                        let chars = str.char_indices();
+
+                        let mut prev = None;
+
+                        for (i, c) in chars {
+                            if prev == None {
+                                prev = Some(c);
+                                continue;
+                            }
```

Comment:
This will cause us to fail to parse `""` correctly, since the trailing `"` will get skipped. I think you should drop this and turn the check below into something like `prev == Some('\\')`.

And then, if you merge the escape check into the quote check, like `c == quote && prev != Some('\\')`, then we don't need any nonlocal control flow except the loop break

Reply:
It doesn't fail to parse since `pos` is never updated in that condition, parsing `""` looks like the following:
- first quote gets captured by `char(quote)`
- `str.char_indices()` has 1 iteration; being the second quote
- the `if prev == None` case triggers since its the first iteration
- loop exits because theres no more elements in the iterable
- `Ok` is returned, and since we didn't update `pos`, its remainder is `"` which is the same as the input.

Reply:
That's only true in isolation, where the `"` is at the end of the input. Consider parsing `["", "foo"]`.
You'll skip the closing `"` on the first string, then carry on until you hit the opening quote of `"` which will be incorrectly interpreted as a close quote.

Reply:
```
_localdev:edgedb> select <array<str>>$0;
Parameter <array>$0: ["", "asdf"] -- array -> Expected ']' in "asdf\"]"
```

Reply:
Adding
```rs
 if c == quote {
     break;
 }

```

into the condition fixes this